##1. Perkenalan

Nama: **Sivi Almanaf Ali Shahab**

Batch: **006**

#2. Problem

Kamu adalah seorang data scientist yang bekerja di salah satu perusahaan manufaktur baterai Lithium ion. Tim produksi meminta kamu untuk mengecek apakah baterai yang mereka buat dalam kondisi bagus atau tidak. Sebelum melakukan pengecekan lebih lanjut, kamu diminta untuk menghitung hambatan/resistansi dan Daya setiap detik:

R=V/I

P=VI

R adalah Kapasitas dalam satuan Ohm
P adalah Daya dalam Watt
I adalah kuat arus dalam satuan Ampere
V adalah tegangan dalam satuan Volt

a. Buatlah dua fungsi eksplisit masing-masing memuat perhitungan hambatan (R) dan daya (P) yang nantinya akan digunakan untuk perhitungan

b. Buat kolom baru dengan nama kolom R/P berisikan hasil perhitungan daya/hambatan dengan kriteria untuk Kapasitas >=2.5 hitung daya, Kapasitas <2.5 hitung nilai hambatan. (Hint: Gunakan for loop dan if statement untuk mengerjakan perintah nomor b, kamu bisa menggunakan list untuk menyimpan perhitungan sementara, setelahnya dapat dimasukkan ke kolom R/P).

c. Hitunglah rata-rata nilai kapasitas untuk arus = 1 Ampere dan cycle ganjil.

## Import Library pandas dan Numpy

In [5]:
import pandas as pd
import numpy as np

##Load Data

Dataset dapat didownload pada link berikut: https://github.com/fahmimnalfrzki/Dataset/blob/main/NMC1-9.xlsx?raw=true

In [6]:
df = pd.read_excel('./data/NMC1-9.xlsx')

# print top 5 from data
df.head()

time(s)    V_m  Current (A)  Cycle       Cap
0        0  4.098        0.997      1  2.840817
1        1  4.096        0.998      1  2.840817
2        2  4.094        0.999      1  2.840817
3        3  4.093        0.998      1  2.840817
4        4  4.092        0.999      1  2.840817

In [7]:
# cek ukuran data
df.shape

(90982, 5)

In [8]:
df.describe()

time(s)           V_m   Current (A)         Cycle           Cap
count  90982.000000  90982.000000  90982.000000  90982.000000  90982.000000
mean   45490.500000      3.607666      0.999264      4.978699      2.806214
std    26264.385433      0.322777      0.001997      2.582135      0.023283
min        0.000000      2.497000      0.993000      1.000000      2.770180
25%    22745.250000      3.422000      0.997000      3.000000      2.786860
50%    45490.500000      3.636000      0.999000      5.000000      2.804878
75%    68235.750000      3.868000      1.001000      7.000000      2.823520
max    90981.000000      4.114000      1.003000      9.000000      2.840817

##Jawaban poin (a)

In [9]:
# definisi fungsi Hambatan dan Daya
def hambatan(V, I):
  return V/I

def daya(V, I):
  return V * I


##Jawaban poin (b)

In [10]:
df_new = df.copy()
rp = []
j=0

for i in df_new['Cap']:
  if i >=2.5:
    rp.append(daya(df_new['V_m'][j], df_new['Current (A)'][j]))
  else:
    rp.append(hambatan(df_new['V_m'][j], df_new['Current (A)'][j]))
  j+=1


df_new['R_P'] = rp
df_new.head()

time(s)    V_m  Current (A)  Cycle       Cap       R_P
0        0  4.098        0.997      1  2.840817  4.085706
1        1  4.096        0.998      1  2.840817  4.087808
2        2  4.094        0.999      1  2.840817  4.089906
3        3  4.093        0.998      1  2.840817  4.084814
4        4  4.092        0.999      1  2.840817  4.087908

In [12]:
# RP = []
# for in range(len(data)):
#   if df_new['Cap'].iloc[i] >= 2.5:
#     RP.append(df_new(data['V_m'].iloc[i],df_new['Current (A)'].iloc[i]))
#   else:
#     RP.append(df_new(data['V_m'].iloc[i],df_new['Current (A)'].iloc[i]))

# data_new['R/P'] = RP
# data_new.head()

In [13]:
df_new.describe()

time(s)           V_m   Current (A)         Cycle           Cap  \
count  90982.000000  90982.000000  90982.000000  90982.000000  90982.000000   
mean   45490.500000      3.607666      0.999264      4.978699      2.806214   
std    26264.385433      0.322777      0.001997      2.582135      0.023283   
min        0.000000      2.497000      0.993000      1.000000      2.770180   
25%    22745.250000      3.422000      0.997000      3.000000      2.786860   
50%    45490.500000      3.636000      0.999000      5.000000      2.804878   
75%    68235.750000      3.868000      1.001000      7.000000      2.823520   
max    90981.000000      4.114000      1.003000      9.000000      2.840817   

                R_P  
count  90982.000000  
mean       3.604989  
std        0.322366  
min        2.490000  
25%        3.420146  
50%        3.633252  
75%        3.864714  
max        4.113109

##Jawaban poin (c)

In [14]:
df_new = df.rename(columns={'Current (A)': 'kuat_arus'})
df_new.columns

Index(['time(s)', 'V_m', 'kuat_arus', 'Cycle', 'Cap'], dtype='object')

In [15]:
df_new.query('kuat_arus==1 and Cycle%2==1')

time(s)    V_m  kuat_arus  Cycle       Cap
15          15  4.085        1.0      1  2.840817
19          19  4.083        1.0      1  2.840817
20          20  4.083        1.0      1  2.840817
21          21  4.082        1.0      1  2.840817
31          31  4.079        1.0      1  2.840817
...        ...    ...        ...    ...       ...
90909    90909  2.640        1.0      9  2.770180
90923    90923  2.615        1.0      9  2.770180
90937    90937  2.589        1.0      9  2.770180
90949    90949  2.565        1.0      9  2.770180
90967    90967  2.528        1.0      9  2.770180

[6830 rows x 5 columns]

In [16]:
df_new.query('kuat_arus==1 and Cycle%2==1')['Cap'].mean()

2.8064845553925855

Rata-rata nilai kapasitas untuk arus = 1 Ampere dan cycle ganjil adalah **2.806485**